src/encoder.rs

```diff
@@ -2704,12 +2704,27 @@ fn encode_partition_topdown<T: Pixel, W: Writer>(
           let match4 = mv_stack.len() > 2 && mv_stack[2].this_mv == mvs[0];
           let match5 = mv_stack.len() > 2 && mv_stack[2].comp_mv == mvs[1];
 
+          let match6 = if mv_stack.len() > 3 {
+            mv_stack[3].this_mv.row == mvs[0].row
+              && mv_stack[3].this_mv.col == mvs[0].col
+          } else {
+            false
+          };
+          let match7 = if mv_stack.len() > 3 {
+            mv_stack[3].comp_mv.row == mvs[1].row
+              && mv_stack[3].comp_mv.col == mvs[1].col
+          } else {
+            false
+          };
```

Comment:
We may simplify this expression. Once we have a complete mapping of modes, we may revisit the preceding expressions as well.
```suggestion
          let match6 = mv_stack.len() > 3 && mv_stack[3].this_mv == mvs[0];
          let match7 = mv_stack.len() > 3 && mv_stack[3].comp_mv == mvs[1];
```